packages/common/src/store/pages/chat/selectors.ts

```diff
@@ -93,6 +93,25 @@ export const getChatMessages = createSelector(
   }
 )
 
+/**
+ * Get the number of unread messages across all chats, including optimistic reads.
+ * Note: This will not reflect an accurate unread messages count, but rather is used
+ * to coalesce count fetched from backend + optimistic reads, to be used in
+ * getHasUnreadMessages.
+ */
+export const getUnreadMessagesCount = createSelector(
+  [getChats, (state: CommonState) => state.pages.chat.unreadMessagesCount],
+  (chats, unreadCount) => {
+    const optimisticUnreadCount = chats.reduce((acc, currentChat) => {
+      return acc + (currentChat.unread_message_count ?? 0)
+    }, 0)
+    return Math.min(unreadCount, optimisticUnreadCount)
```

Comment:
On thinking further...

- `markChatAsRead` sets `unread_message_count` to 0
- `incrementUnreadCount` adds 1 to `unread_message_count`

but neither of those right now impact the `unreadCount` pulled from the new SDK endpoint. That means:

- if `unreadCount` is 0, but we get a new message in that session, optimisticUnreadCount will be 1 but this method returns zero
- if `unreadCount` is positive, but we optimistically mark every chat _loaded_ as read, optimistic read count will be 0 and the method returns zero (but `unreadCount` may still be counting unreads for chats we haven't paginated through yet, there may still be unreads)

So unfortunately I don't think we can take the min or max here, I think we do need to independently update the store for `unreadMessagesCount` in `incrementUnreadCount` and `markAsRead`

Reply:
I am incrementing global `unreadMessagesCount` in `incrementUnreadCount`, so that takes care of the first case. But yeah I think you're right in 2nd case.